### ISSUE TITLE
Firefox requires xpi file type for extension code, but doesn't allow it for extension source code

### ISSUE THREAD

**Issue by jellyfith** [extension-js/extension.js#229]
When building for firefox, we automatically compress results into a XPI file. This is good for the app code as it is preferred for when submitting the code to firefox.

The problem is that firefox also requires the submission of the source code for extensions that use bundlers. That source code needs to be compressed into any of the following formats: .zip, .tar.gz, .tgz, .tar.bz2

We already provide the ability to zip-source, we should add a way to specify the file type for posterity.

**Comment by bo-carey** [comment]
If anyone needs a workaround for this, you can zipSource for your chrome build and then just zip without source for firefox. That way you get the source zip in .zip format.

**Comment by cezaraugusto** [comment]
hey @bo-carey, not sure I'm following. what is the optimal workflow here in your opinion?

you can compress the final build via `--zip` flag and the source via `--zip-source` like `pnpm extension build --zip`.

**Comment by bo-carey** [comment]
Pretty much that! I wasn't sure what you might think of the issue. I'd be happy to take this on as a first issue if that's alright.

**Comment by cezaraugusto** [comment]
sure! you can use `extension build --zip` to zip your distribution files and `extension build --zip-source` to zip the source files. you can also combine both:

```
# Generates a zip artifact for both production and development
extension build --zip-source --zip
```

let me know if that's not the case, happy to help

**Comment by jellyfith** [comment]
Yeah, I noticed that if I simply run `extension build --all --zip --zip-source` instead of targeting firefox only, I get the source as a .zip and the firefox compressed to .xpi like I needed. We can close this now :sweat_smile:

**Comment by jellyfith** [comment]
Yeah, I noticed that if I simply run `extension build --all --zip --zip-source` instead of targeting firefox only, I get the source as a .zip and the firefox compressed to .xpi like I needed. We can close this now :sweat_smile:
